src/pkg/packager/publish.go

```diff
@@ -26,7 +26,7 @@ import (
 // Publish publishes the package to a registry
 func (p *Packager) Publish() (err error) {
 	_, isOCISource := p.source.(*sources.OCISource)
-	if isOCISource {
+	if isOCISource && p.cfg.PublishOpts.SigningKeyPath == "" {
```

Comment:
its a bit more effort, but we _can_ use the filtering behavior of our copy fn to not copy over the existing sig, then pull down the `zarf.yaml`, then re-sign, then push those 2 layers back up, then make a new manifest and push. If we wanted to go for max efficiency that would do the trick, but it does introduce some complexity and needed understanding of the push/pull flow.

Reply:
added a comment to the component layers tech debt - it would be more efficient but wanted to fix things faster